### ISSUE TITLE
[DV] How to get started?

### ISSUE THREAD

**Issue by yoursunny** [named-data/ndnd#90]
To deploy a network with **ndnd dv** routing daemon:

* What keys and certificates are required, and where to put them?
* How to configure neighbor routers?
* Is it necessary to manually create faces and strategies?
* How to announce and withdraw prefixes?

**Comment by pulsejet** [comment]
Some basic docs now in place. There's no security at the moment

To use DV, just start two instances on different forwarders, then follow docs here to add neighbor links.
https://github.com/named-data/ndnd/blob/main/tools/dvc/README.md

No further config is needed.

To announce prefix, register route to NFD with origin=65

**Comment by yoursunny** [comment]
> There's no security at the moment

Looks like I'd hold off deployment until security validation is in place.

**Comment by pulsejet** [comment]
Yes, this isn't production code by any means yet. Though if you have a test setup and can give it a trial run that'll be very helpful

**Comment by yoursunny** [comment]
DV has a keychain option now:
https://github.com/named-data/ndnd/blob/64d1486b08e060c38d98da6ad16a3fc88337562b/dv/dv.sample.yml#L12-18

What should I put in the keychain?

**Comment by pulsejet** [comment]
Please wait, this is still wip

**Comment by pulsejet** [comment]
An example is now available at https://github.com/named-data/ndnd/blob/main/docs/daemon-example.md

Still rough around the edges and subject to change

**Comment by yoursunny** [comment]
https://github.com/named-data/ndnd/blob/c542c2a7579383e5fd44462d93d80bf64a6c2f8b/docs/daemon-example.md?plain=1#L119-L121
https://github.com/named-data/ndnd/blob/c542c2a7579383e5fd44462d93d80bf64a6c2f8b/docs/daemon-example.md?plain=1#L130-L132

I don't see a field that specifies the filename of own private key.
How does DV instance know which file is its own private key?
Or is it the expectation that exactly one private key is present in the folder?

**Comment by pulsejet** [comment]
The trust schema is used to identify usable private keys. If multiple keys are matching, the one with the newest available certificate (by version) is used.

**Comment by yoursunny** [comment]
DV with security is finally working as of 1cf08f0d27d259c7b54fa872a8a0a5d070c9bbc9, using https://github.com/named-data/ndnd/issues/115#issue-2811595850 configuration.

Retrieval is very slow and eventually fails:
```text
$ time docker run --rm --cap-drop ALL --network none --user $(id -u):$(id -g) -v run-ndn:/run/nfd docker.yoursunny.dev/ndnd cat /yoursunny/vno/1.bin >1.bin
time=2025-01-26T18:04:48.850Z level=FATAL msg="Error fetching object" tag=cat err="consume: fetch seg failed with result: Timeout"

real	21m51.307s
user	0m0.147s
sys	0m0.974s
```

In contrast, ndncatchunks is reasonably fast (producer and forwarders are still ndnd):
```text
$ time docker run --rm --cap-drop ALL --network none -v run-ndn:/run/nfd ghcr.io/named-data/ndn-tools ndncatchunks /yoursunny/vno/1.bin >2.bin


All segments have been received.
Time elapsed: 230.131 seconds
Segments received: 134218
Transferred size: 1.07374e+06 kB
Goodput: 37.326309 Mbit/s
Congestion marks: 24 (caused 22 window decreases)
Timeouts: 714 (caused 53 window decreases)
Retransmitted segments: 709 (0.525469%), skipped: 5
RTT min/avg/max = 117.819/122.114/203.534 ms

real	3m50.580s
user	0m1.686s
sys	0m3.869s
```

**Comment by pulsejet** [comment]
Likely because there's no congestion control (#97)

> ndncatchunks is reasonably fast

That still looks pretty awful. By any chance, any numbers with NFD and/or tcp face?
